ext/nnstreamer/tensor_filter/tensor_filter_armnn.cc

```diff
@@ -164,8 +165,37 @@ int
 ArmNNCore::makeCaffeNetwork (std::map < std::string,
     armnn::TensorShape > &input_map, std::vector < std::string > &output_vec)
 {
-  /** @todo fill this */
-  return -EPERM;
+  bool unknown_input_dim = false;
+
+  armnnCaffeParser::ICaffeParserPtr parser =
+      armnnCaffeParser::ICaffeParser::Create ();
+
+  for (auto const &inputs:input_map) {
+    if (inputs.second.GetNumDimensions () == 0) {
+      unknown_input_dim = true;
+    }
+  }
+
+  if (unknown_input_dim) {
+    network = parser->CreateNetworkFromBinaryFile
+      (model_path, {}, output_vec);
+  } else {
+    network = parser->CreateNetworkFromBinaryFile
+      (model_path, input_map, output_vec);
+  }
+
+  /** set input/output bindings */
+  for (auto const &output_name:output_vec) {
+    outputBindingInfo.push_back (
+        parser->GetNetworkOutputBindingInfo (output_name));
+  }
+
+  for (auto const &inputs:input_map) {
```

Comment:
Do they have valid values when `unknown_input_dim` is true,

Reply:
When `unknown_input_dim` is true, the dimension of the inputs are missing (`input_map.second`).
However, here we want to extract just the input names (`input_map.first`).